subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -1714,10 +1714,7 @@ u32_t ticker_start(u8_t instance_index, u8_t user_id, u8_t ticker_id,
 
 	user = &instance->user[user_id];
 
-	last = user->last + 1;
-	if (last >= user->count_user_op) {
-		last = 0U;
-	}
+	last = (user->last + 1) % user->count_user_op;
```

Comment:
Won't this result in [a division instead of that conditional set to 0](https://community.arm.com/developer/ip-products/processors/b/processors-ip-blog/posts/divide-and-conquer)?
On a NRF51 (Cortex M0) that would probably be quite a bit of overhead.

Reply:
You are right. The compiler inserts a branch to a routine in software that does the division resulting in worse performance. Good catch!